### PR TITLE
Fix for `resolve_fraction_unit` zero division error (#2502)

### PR DESCRIPTION
**Pull request by davep**
This will require some serious review by especially @willmcgugan (as function author) and @rodrigogiraoserrao (as maths wizard), as I'm not confident I really follow what the code is doing. However, having hand-debugged this I *think* this, or something similar, is the solution to #2502.

The approach here is that there's no point in carrying on looping around if there is no remaining fraction, and on the way out, if the remaining space has gone negative, the fallback value (`Fraction(1)`) should be used (before now it was if the remaining space was `0`, so this is in effect clamping the lower bound).

### CODE REVIEW

**Comment by rodrigogiraoserrao** [comment]
Your fix doesn't seem to have mathematical issues 😛 
But your fix also does beg the question about whether there might be a configuration that would take the `remaining_space`/`remaining_fraction` to be negative values during the loop, which wouldn't break the loop.

Hence, my comment in the review.

**Comment by davep** [comment]
Breaking unit test has now been simplified. It's now been reduced down to a single widget that is `1fr` and which has a `min-width` bigger than the incoming `remaining_space`.

**Review by rodrigogiraoserrao — APPROVED** [review]
I'm happy with this.

File: src/textual/_resolve.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def resolve_scalar(
 
     return (
         Fraction(remaining_space, remaining_fraction)
-        if remaining_space
+        if remaining_space > 0
```
Comment (darrenburns) [inline src/textual/_resolve.py:169]: Should this be `remaining_fraction > 0`?

**Pull request merged**